include/envoy/upstream/cluster_manager.h

```diff
@@ -73,6 +73,15 @@ class ClusterManagerFactory;
 /**
  * Manages connection pools and load balancing for upstream clusters. The cluster manager is
  * persistent and shared among multiple ongoing requests/connections.
+ * Cluster manager is initialed in two phases. In the first phase which begins at the construction
+ * all primary (i.e. not provisioned through xDS) clusters are initialized.
+ * After the first phase the RTDS (if configured) initialization begins. This allows runtime
```

Comment:
+1

Reply:
I wanted to avoid leaking overall initialization order into the cluster manager. So that is why I put 2 phases there. 

1. In the first phase primary cluster are brought up.
1. The server does something else, which cluster manager does not need to care about.
1. Then the second phase begins where secondary clusters are initialized.

From the cluster manager perspective there are two phase only. I've updated comment and moved most of it into the InstanceImpl where the order is (mostly) established.